### PR TITLE
feature: add payPalCampaigns to CustomerSessionRequest

### PR DESCRIPTION
**Pull request by santugowda**
### Summary of changes
- Add `ShopperInsightsCampaign(id)` model and an optional `payPalCampaigns` list on `CustomerSessionRequest`.
- Build the `paypalCampaigns` JSON array (`[{ id }]`) in `CustomerSessionRequestBuilder` and expose it on `JsonRequestObjects`.
- Foundation only — the create/update session and generate-recommendations APIs are wired to actually send `paypalCampaigns` in the follow-up PR #1649 (only ordering constraint: #1647 → #1649).

Part of enabling PayPal co-marketing campaign attribution (TikTok first) through Shopper Insights v2. Backend is merged (cosmos-apps #10542). The v2 surface is `@ExperimentalBetaApi`.

**Steps to test:** unit tests cover the builder (`payPalCampaigns` present / empty / null). No runtime payment-flow change in this PR — the field is inert until wired in #1649, where it is verified end-to-end.

### AI Usage

**Which AI Agent Was Used?**
- [ ] Copilot 
- [x] Claude 
- [ ] Other (Type Name Here)

**How was AI used?**
Code generation for the model, request builder, and unit tests; research across the PRD/HLD and backend PRs to confirm the wire contract.

**Estimated AI Code Contribution**
- [ ] less than 30% 
- [ ] 30 - 60% 
- [x] 60 - 100% 

### Checklist
- [x] Added a changelog entry
- [x] Tested and confirmed payment flows affected by this change are functioning as expected

- santugowda

----
### Inner Source Process
Internal to PayPal contributors should fill out this section. All others can delete.

PR should follow these steps before codeowners review will begin:
1. Comment `/inner source` on this PR — this will automatically add the `inner source` and `tech lead review required` labels. Open the PR in a draft state.
2. PR should be reviewed by and approved by your team's technical lead, we do not allow LGTM reviews, there should be comments and feedback provided on all PR reviews
3. Once the above steps are completed, comment `/ready` on this PR — this will automatically remove the `tech lead review required` label. Move the PR to ready to review.
4. PR comments must be addressed within 24 hours, if you are unable to address within this timeframe, move the PR back to a draft state so our team knows not to review

### Inner Source Checklist
- [x] Added all labels to the PR
- [x] Provide steps to test the flows changed, if applicable in the summary
- [ ] Demo video of the functionality, if applicable
- [x] All upstream dependencies are merged in and this PR can be released at any time; PRs should not be opened until this is true
- [x] Unit tests and builds have been run locally and pass/compile as expected

### CODE REVIEW

File: ShopperInsights/src/test/java/com/braintreepayments/api/shopperinsights/v2/internal/CustomerSessionRequestBuilderUnitTest.kt [hunk 1]
```diff
@@ -76,5 +77,34 @@ class CustomerSessionRequestBuilderUnitTest {
...
+        val expectedCampaigns = JSONArray().apply {
+            put(JSONObject().put("id", "campaign-1"))
+            put(JSONObject().put("id", "campaign-2"))
+        }
+
+        JSONAssert.assertEquals(expectedCampaigns, result.payPalCampaigns, true)
+    }
+
+    @Test
+    fun `createRequestObjects returns null payPalCampaigns when list is empty`() {
```
Comment (anibalb2500) [inline ShopperInsights/src/test/java/com/braintreepayments/api/shopperinsights/v2/internal/CustomerSessionRequestBuilderUnitTest.kt]: Should we also have a test passing in null?

**Review by anibalb2500 — COMMENTED** [review]
Small nitpick but looks good.

**Comment by anibalb2500** [comment]
LGTM

**Comment by jaxdesmarais** [comment]
@santugowda can 1 of these be merged in without the others? There are 3 PRs and it seems like all need to be merged/released at once. If that is the case please use the workflow to create a feature branch and open 1 PR at a time.

It looks like the AI you used also removed a lot of the checklist. I recommend opening the PR manually to ensure you're meeting all criteria for review.

**Comment by santugowda** [comment]
> @santugowda can 1 of these be merged in without the others? There are 3 PRs and it seems like all need to be merged/released at once. If that is the case please use the workflow to create a feature branch and open 1 PR at a time.
> 
> It looks like the AI you used also removed a lot of the checklist. I recommend opening the PR manually to ensure you're meeting all criteria for review.

@jaxdesmarais 
On merging: they don't all need to release at once. #1648 (PayPal checkout) is fully independent. 
#1647 (Shopper Insights foundation) can also merge on its own it only adds an optional, `@ExperimentalBetaApi `ffield that isn't wired up until #1649. So the only ordering constraint is #1647 → #1649.

 IMO don't need a feature branch, but I'm happy to work either way and get this into a feature branch. 

Regarding checklist - the generated description dropped several template items let me fix it.

**Comment by jaxdesmarais** [comment]
> IMO don't need a feature branch, but I'm happy to work either way and get this into a feature branch.

These all relate to 1 feature (`payPalCampaigns`) and therefore should be pointed to a feature branch. It would not make sense to release 1 piece without the others. Also releasing something that isn't hooked up properly does not accurately represent ready for release as it would mean it could be live and not work in the SDK.

I also expect there should be a companion PR for this feature on iOS?

**Comment by santugowda** [comment]
> These all relate to 1 feature (payPalCampaigns) and therefore should be pointed to a feature branch. 

Yes - but #1648 is for rest api /create orders, feature branch workflow is an over kill here. 
Other two is wrt shopper insight - that uses Graph QL so different flows, this can use feature branch workflow. 

Edit * : Update the PR to point to feature branch @jaxdesmarais

File: CHANGELOG.md [hunk 2]
```diff
@@ -30,6 +30,7 @@
     * Expose `createCustomerSession` as a public suspend function
     * Expose `updateCustomerSession` as a public suspend function
     * Expose `generateCustomerRecommendations` as a public suspend function
+    * Add `CustomerSessionRequest.campaigns` to associate PayPal campaigns with a customer session
```
Comment (saralvasquez) [inline CHANGELOG.md]: We've had a release recently. Could you move this under the `unreleased` section of the changelog?

**Review by saralvasquez — COMMENTED** [review]
The CHANGELOG entry needs to be moved, but after that I think this looks good to go in

**Comment by saralvasquez** [comment]
Oh weird, for some reason the `Files Changed` comparison isn't showing what the changelog for the feature branch actually looks like 🤔

**Comment by saralvasquez** [comment]
Honestly it'll probably be easier to just not add the changelog entries until the end when we merge the feature branch into main. That way you don't have to keep updating them

**Comment by santugowda** [comment]
@saralvasquez  - as requested I have removed the CHANGELOG. Will add it later to this feature branch

**Review by jaxdesmarais — APPROVED** [review]
Thanks for updating per the updated LLD. 🚀

File: CHANGELOG.md (hunk 2 above)
Comment (santugowda) [reply CHANGELOG.md]: removed changelog for now, will add it as part of new commit in a different PR pointing to this feature branch

**Comment by santugowda** [comment]
@saralvasquez / @jaxdesmarais  - can we merge this.

**Review by saralvasquez — APPROVED** [review]
🎉

**Review by saralvasquez — APPROVED** [review]
Not sure where we are at with the iOS logic discussion. But once we have alignment I think this is good to go in

**Comment by jaxdesmarais** [comment]
I am going to merge this in as I believe this is the intended design and separation of responsibility. That said, @santugowda depending on the outcome of the iOS discussion we may need to open a small PR to the feature branch on Android. I don't want to block this going in though as it follows existing SDK patterns.

**Comment by santugowda** [comment]
Understood, thanks @jaxdesmarais . 
I will reopen #1649  PR which is follow up to this.
